src/qt/signverifymessagedialog.cpp

```diff
@@ -133,30 +133,31 @@ void SignVerifyMessageDialog::on_signMessageButton_SM_clicked()
         return;
     }
 
-    CKey key;
-    if (!model->wallet().getPrivKey(GetScriptForDestination(destination), CKeyID(*pkhash), key))
-    {
-        ui->statusLabel_SM->setStyleSheet("QLabel { color: red; }");
-        ui->statusLabel_SM->setText(tr("Private key for the entered address is not available."));
-        return;
+    std::string message = ui->messageIn_SM->document()->toPlainText().toStdString();
+    std::string sig;
+    SigningError err = model->wallet().signMessage(message, *pkhash, sig);
+
+    QString error;
+    switch (err) {
+        case SigningError::OK:
+            error = tr("No error");
+        case SigningError::PRIVATE_KEY_NOT_AVAILABLE:
+            error = tr("Private key for the entered address is not available.");
```

Comment:
Missing `brake;`?

Reply:
Also these strings can be obtained from`SigningErrorString` in `error.cpp` (the wording here seems better)

Reply:
The strings from `SigningErrorString` won't be translated (we don't translate things for RPC), so that logic had to be duplicated here. Also, the strings are different to maintain compatibility.

Reply:
Added the breaks in.